core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/CompactExpressionEvaluator.scala

```diff
@@ -198,14 +199,33 @@ class CompactExpressionEvaluator(repository: CompactEntityRepository) extends Ex
                 }
 
                 // Pre-compute a map from expression -> ExpressionAndResult for O(1) lookup
-                val expressionToResultMap: Map[String, Seq[ExpressionAndResult]] =
+                val entityExpressionToResultMap: Map[String, Seq[ExpressionAndResult]] =
                   combinedExpressionAndResults.groupBy { case (expression, _) => expression }
 
+                // Transform workspaceExpressionResults to the same shape and add to the map
+                val workspaceExpressionAndResults: Seq[ExpressionAndResult] = workspaceExpressionResults.map {
+                  case (lookupExpression, tryValues) =>
+                    // Create a result map with the workspace values for all root entity names
+                    val resultMap = rootEntityNames.map { entityName =>
+                      entityName -> tryValues
+                    }.toMap
+
+                    (lookupExpression, resultMap)
+                }.toSeq
+
+                val workspaceExpressionToResultMap: Map[String, Seq[ExpressionAndResult]] =
+                  workspaceExpressionAndResults.groupBy { case (expression, _) => expression }
+
+                // Combine both maps, with workspace results taking precedence if there are conflicts
+                val expressionToResultMap: Map[String, Seq[ExpressionAndResult]] =
+                  entityExpressionToResultMap ++ workspaceExpressionToResultMap
+
```

Comment:
do workspace results take precedence in the existing code?

Reply:
I don't know! It's very hard to figure out what's going on in the existing code 😅 But I think we wouldn't expect there to be conflicts, or that it wouldn't matter, since the key to the map is the actual expression, i.e. `workspace.attribute` or `this.bam`, so the same key should give the same result, and you wouldn't expect a non-workspace expression to be in the non-workspace map and vice versa